eth/stagedsync/stage_headers.go

```diff
@@ -147,6 +148,18 @@ func HeadersPOS(
 	cfg HeadersCfg,
 	useExternalTx bool,
 ) error {
+	executionProgress, err := stages.GetStageProgress(tx, stages.Execution)
+	if err != nil {
+		return err
+	}
+	// We prevent pos reverse downloading from happening concurrently with an interrupted sync.
+	if s.BlockNumber > executionProgress {
```

Comment:
Hmm. Not sure about this. In the PoW case we don't wait for execution to finish but rather download new headers first, right?

Reply:
it is less code changes, i dont think it makes too much of a difference.

Reply:
we would have to pass the headerdownload to all other stages, for now i would reduce this PR only to this change, if you are okay with it